docs/versioning.rst

```diff
@@ -0,0 +1,89 @@
+**********
+Versioning
+**********
+
+
+Current Status
+==============
+
+.. list-table:: Releases
+   :widths: 20 30 30 20
+   :header-rows: 1
+
+   * - Release
+     - Supported Runtimes
+     - Minimum Trace Agent
+     - Status
+   * - 0.x
+     - CPython: 2.7, 3.5+
+     - 
+     - MAINTENANCE [2022-02-25]
+   * - 1.x
+     - CPython: 2.7, 3.5+
+     - 7.28
+     - CURRENT
+
+
+The **status** of each release determines when changes are made:
+
+MAINTENANCE
+    No longer under development.
+
+    Receives bug fixes and security fixes.
+
+CURRENT
+    Under active development.
+
+    Receives bug fixes and security fixes.
+
+    Receives backwards compatible features.
+
+DEVELOPMENT
+    Under active development.
+
+    Receives backwards incompatible features.
+
+    Will become the next CURRENT release.
+
+
+Interface
+=========
+
+The definition of the **internal** interface:
```

Comment:
We might need to explicitly state that breaking changes can be made to the internal interface at any time without warning or a major version bump.

Might be good to cover this in the "Versions" section, and then move this Interface section to the end?